packages/framework/src/validators/zod.validator.ts

```diff
@@ -30,6 +29,9 @@ export class ZodValidator implements Validator<ZodSchema> {
 
   transformToJsonSchema(schema: ZodSchema): JsonSchema {
     try {
+      // eslint-disable-next-line global-require
+      const zodToJsonSchema = require('zod-to-json-schema') as typeof import('zod-to-json-schema').default;
```

Comment:
Just out of curiosity, should we maybe try async imports instead of require?

Reply:
`transformToJsonSchema` needs to be completely synchronous because it's transitively used during discovery phase, which is completely synchronous too. This ensures that you can use methods on the defined workflow without needing to `await` the `workflow(...)` definition, such as:

```typescript
export const myNamedWorkflow = workflow('welcome-onboarding', async ({ step }) => {
    await step.email('send-email', async () => {
        return ({
            subject: 'Welcome to our platform!',
            body: 'Welcome to our platform!',
        })
    });
}, {
    payloadSchema: {
        type: 'object',
        properties: {
            name: { type: 'string' },
        },
        required: ['name'],
        additionalProperties: false,
    } as const,
});

await myNamedWorkflow.trigger({
    payload: {
        name: 'John Doe'
    },
    to: '+1234567890'
})
```

the alternative (bad DX) would be:
```typescript
await (await myNamedWorkflow).trigger({
    payload: {
        name: 'John Doe'
    },
    to: '+1234567890'
})
```

Reply:
Thanks for the explanation